app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -11,53 +16,79 @@
 
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand = EMPTY_STRING;
-    private String secondOperand = EMPTY_STRING;
+    private Operand firstOperand = new Operand();
+    private Operand secondOperand = new Operand();
     private String operator = EMPTY_STRING;
     private String result = EMPTY_STRING;
 
-
     @Override
     public void saveNumber(String number) {
         if (operator.isEmpty()) {
-            firstOperand += number;
+            firstOperand.addNumber(number);
         } else {
-            secondOperand += number;
+            secondOperand.addNumber(number);
         }
     }
 
     @Override
     public String getPartialResult() {
+        return String.valueOf(firstOperand.getValue()) + operator + String.valueOf(secondOperand.getValue());
+    }
 
-        return (firstOperand + operator + secondOperand);
+    private boolean isValidOperation() {
+        if (operator.isEmpty()) {
+            if (firstOperand.isEmpty()) {
+                result = EMPTY_STRING;
+            } else {
+                result = String.valueOf(firstOperand.getValue());
+            }
+            return false;
+        } else if (secondOperand.isEmpty()) {
+            result = ERROR_MESSAGE_INVALID_FORMAT;
+            return false;
+        }
+        return true;
     }
 
     @Override
     public String getFullResult() {
 
-        switch (operator) {
-
-            case OPERATOR_SUM:
-                if (result == EMPTY_STRING) {
-                    result = String.valueOf(Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand));
-                }
-
-                break;
-
-            case OPERATOR_MINUS:
-                result = String.valueOf(Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand));
-                break;
-
-            case OPERATOR_DIVIDE:
-                result = String.valueOf(Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand));
-                break;
-
-            case OPERATOR_MULTIPLY:
-                result = String.valueOf(Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand));
-                break;
-
-            default:
-                result = ERROR_MESSAGE;
+        if (isValidOperation()) {
+
+            switch (operator) {
+
+                case OPERATOR_SUM:
+                    result = String.valueOf(firstOperand.getValue() + secondOperand.getValue());
+                    break;
+
+                case OPERATOR_MINUS:
+                    result = String.valueOf(firstOperand.getValue() - secondOperand.getValue());
+                    break;
+
+                case OPERATOR_DIVIDE:
+                    if (firstOperand.getValue() == secondOperand.getValue()) {
```

Comment:
this condition are right? because if you try to do 7 / 7 = 1

Reply:
You were right, now its fixed